File: javascript/gigasecond/gigasecond.js
GIGAMILLI = Math.pow(10, 9) * 1000;

var Gigasecond = function(from) {
    this.from = from.getTime();
};

Gigasecond.prototype.date = function() {
    return new Date(GIGAMILLI + this.from);
};

module.exports = Gigasecond;

File: go/gigasecond/gigasecond.go
package gigasecond

import (
	"math"
	"time"
)

const testVersion = 4

var gigasecond = int64(math.Pow(10, 9))

func AddGigasecond(t time.Time) time.Time {
	return time.Unix(t.Unix()+gigasecond, 0).UTC()
}
